addons/project/project.py

```diff
@@ -434,6 +434,10 @@ def create(self, cr, uid, vals, context=None):
         return project_id
 
     def write(self, cr, uid, ids, vals, context=None):
+        for record in self.browse(cr, uid, ids, context=context):
+            if 'active' in vals and not vals.get('active') and ('state' in vals and vals.get('state') in ('open', 'pending', 'template') or 'state' not in vals and record.state in ('open', 'pending', 'template')):
```

Comment:
Are you sure there will be a single record for write ? I used for loop to raise UserError if any of the record is in Open, Pending or Template state, consider the scenario of Listview, I select more than 4 records where 2 records in Open state, and then click Archive from the sidebar, it should not allow to archive records.

Please recheck the line below, I used record.state in ('open', 'pending', 'template') condition:
if 'active' in vals and not vals.get('active') and ('state' in vals and vals.get('state') in ('open', 'pending', 'template') or 'state' not in vals and record.state in ('open', 'pending', 'template')):